generators/example/src/main/scala/Top.scala

```diff
@@ -101,3 +103,12 @@ class TopWithInitZero(implicit p: Parameters) extends Top
 class TopWithInitZeroModuleImp(l: TopWithInitZero) extends TopModule(l)
   with HasPeripheryInitZeroModuleImp
 // DOC include end: TopWithInitZero
+
+class TopWithIceNIC(implicit p: Parameters) extends Top
+    with HasPeripheryIceNIC {
+  override lazy val module = new TopWithIceNICModule(this)
+}
+
+class TopWithIceNICModule(outer: TopWithIceNIC)
+  extends TopModule(outer)
+  with HasPeripheryIceNICModuleImp
```

Comment:
This should be refactored as `CanHavePeripheryIceNICModuleImp`, but I guess I can do that in the future.

Reply:
The BlockDevice doesn't use CanHave* either currently. We should just do all of them in one go later on.